src/lib/react-server-route.js

```diff
@@ -0,0 +1,109 @@
+/*eslint-disable no-unused-vars*/
+import React from 'react';
+/*eslint-enable no-unused-vars*/
+
+import url from 'url';
+import queryParser from './queryParser.js';
+var langmap = require("langmap");
+var HTML = require('../pages/index.js');
+
+function routeFileContent(locales) {
+  var locationParser = require('./location-parser.js')(langmap, locales);
+  return function(request, reply) {
+    var reactDOM = require("react-dom/server");
+    var renderToString = reactDOM.renderToString;
+    var renderToStaticMarkup = reactDOM.renderToStaticMarkup;
+    var routes = require('../routes.js');
+    var reactRouter = require('react-router');
+    var match = reactRouter.match;
+    var RouterContext = reactRouter.RouterContext;
+    var getMessages = require('./get-messages.js');
+    var CreateElement = require('../components/create-element.js');
+
+    var location = url.parse(request.url).pathname;
+    location = encodeURI(location);
+    var search = url.parse(request.url).search || "";
+    var parsedLocation = locationParser(request.headers["accept-language"], location);
+    var parsedRedirect = parsedLocation.redirect;
+    var locale = parsedLocation.locale;
+
+    function generateHTML(renderProps) {
+      var messages = getMessages(locale);
+      var favicon = "/assets/images/favicon.8af3a74ede48e250ceb935c026242483.ico";
+      var twitterImage = "/assets/images/EOY_Twitter_v8_EN.d1bb5d2a5ce35859d038df852d9e6a0a811beaac.png";
+      var facebookImage = "/assets/images/EOY_facebook_v1.a152496406bad899d1a920f6d6b9f507.png";
```

Comment:
Should just file these. I suspect though, once we get started we can just remove these and replace with share progress.